versions/1/data_stream/fields/fields.spec.yml

```diff
@@ -6,66 +6,82 @@ spec:
   type: array
   items:
     type: object
+    # TODO: We should disable additionalProperties. There are properties used in
+    # elastic/integrations that have no purpose or are typos.
     additionalProperties: true
     properties:
       name:
-        description: Name of field
-        type: string
-      title:
-        description: Title of field
+        description: >
+          Name of field. Names containing dots are automatically split into
+          sub-fields.
         type: string
+        # TODO: Should we allow wildcards if they are not implemented in Fleet?
+        pattern: '^[_@A-Za-z0-9]+(\.[_@A-Za-z0-9]+)*$'
       type:
         description: Datatype of field
         type: string
         enum:
         - alias
-        - histogram
-        - constant_keyword
-        - text
-        - match_only_text
-        - keyword
-        - long
-        - integer
-        - short
+        - array
+        - binary
+        - boolean
         - byte
-        - double
-        - float
-        - half_float
-        - scaled_float
+        - constant_keyword
         - date
         - date_nanos
-        - boolean
-        - binary
-        - integer_range
-        - float_range
-        - long_range
-        - double_range
         - date_range
-        - ip_range
-        - group
+        - double
+        - double_range
+        - flattened
+        - float
+        - float_range
         - geo_point
-        - object
+        - group
+        - half_float
+        - histogram
+        - integer # Fleet always maps this as 'long'.
+        - integer_range
         - ip
+        - ip_range
+        - keyword
+        - long
+        - long_range
+        - match_only_text
         - nested
-        - array
-        - flattened
-        - wildcard
-        - version
+        - object
+        - scaled_float
+        - short
+        - text
         - unsigned_long
+        - version
+        - wildcard
```

Comment:
I think there was some kind of "order" in this enum (numbers together, strings together...). I am ok though with using alphabetical order, but maybe we can leave the reordering of this list for a different PR, so it is clearer here if any type is being added or removed.

Reply:
We can consider sorting and do that in another PR. I have reverted that change.